src/Map.js

```diff
@@ -743,12 +747,12 @@ export function deepMerger(existing, value, key) {
     is(existing, value) ? existing : value;
 }
 
-export function deepMergerWith(merger) {
+export function deepMergerWith(merger, keyPath = []) {
   return (existing, value, key) => {
     if (existing && existing.mergeDeepWith && isIterable(value)) {
-      return existing.mergeDeepWith(merger, value);
+      return existing.mergeDeepWith([...keyPath, key], merger, value);
```

Comment:
`keyPath.concat([ key ])` would result in a much cleaner built JS file, but the same behavior you're looking for.